### PR TITLE
fix(caldav): Correctly handle calendar recreation for invitations when the current calendar is in the trashbin

### LINKED ISSUE #4098: Calendar invitations end up in the trash bin, if the personal calendar had been deleted
### Steps to reproduce

1. Create a calendar *personal* if you don't have one already (the URI *might* actually have to be `personal`)
2. Delete the calendar again, but leave it in the trash bin (no permanent delete)
3. Create a new personal calendar (the URI will be `personal-1`)
4. Invite yourself to an event from another user's calendar

### Expected behavior

The event I'm invited to shows up in my *new* personal calendar.

### Actual behaviour

The event doesn't show up, because the backend picks `personal` despite the deletion. Once I restore the first personal calendar, I'll see the event.

### Calendar app version

3.2

### CalDAV-clients used

_No response_

### Browser

_No response_

### Client operating system

_No response_

### Server operating system

_No response_

### Web server

_No response_

### Database engine version

_No response_

### PHP engine version

_No response_

### Nextcloud version

_No response_

### Updated from an older installed version or fresh install

_No response_

### List of activated apps

_No response_

### Nextcloud configuration

_No response_

### Web server error log

_No response_

### Log file

_No response_

### Browser log

_No response_

### Additional info

The bug happens in `\OCA\DAV\CalDAV\Schedule\Plugin::propFindDefaultCalendarUrl`.

https://github.com/nextcloud/calendar/issues/1835 *could* help with this.

### PR DESCRIPTION
**Pull request by tcitworld**
Follow-up to #32361, closes https://github.com/nextcloud/calendar/issues/4098

## TODO

- [x] fix tests

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/CalDAV/Schedule/Plugin.php [hunk 1]
```diff
@@ -329,12 +328,12 @@ public function propFindDefaultCalendarUrl(PropFind $propFind, INode $node) {
...
 				$calendarHome = $this->server->tree->getNodeForPath($calendarHomePath);
-				if (!$calendarHome->childExists($uri)) {
+				$currentCalendarDeleted = false;
+				if (!$calendarHome->childExists($uri) || $currentCalendarDeleted = $this->isCalendarDeleted($calendarHome, $uri)) {
 					// If the default calendar doesn't exist
 					if ($isResourceOrRoom) {
-						$calendarHome->getCalDAVBackend()->createCalendar($principalUrl, $uri, [
-							'{DAV:}displayname' => $displayName,
-						]);
+						// Resources or rooms can't be in the trashbin, so we're fine
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Schedule/Plugin.php:335]: **question:** Is this assumption valid @ChristophWurst @miaulalala ?

**Review by miaulalala — APPROVED** [review]
LGTM

**Review by st3iny — APPROVED** [review]
Tested and works.

**Comment by st3iny** [comment]
@tcitworld This PR fully addresses https://github.com/nextcloud/calendar/issues/4098, right? We should link the issue then.

**Pull request merged**